iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/DockerNodeBuilder.java

```diff
@@ -76,16 +77,45 @@ public Object createTerminal(Input input, int startIndex, int endIndex, List<Tri
     return new SyntaxTokenImpl(value, range, getCommentsForToken(range));
   }
 
+  /**
+   * Compute the {@link TextRange} from a value at the given {@code startIndex}, regarding the provided {@code input} (source code).
+   * In case it extend to multiple lines, we build a {@link CompoundTextRange} object with a reference to each line {@link TextRange}.
+   * This is required to allow to track back original source code before preprocessing, particularly needed when raising issue
+   * on the preprocessed source code.
+   */
   protected TextRange tokenRange(Input input, int startIndex, String value) {
```

Comment:
This method algorithm should be improved in a second step: it would be better to be able to determine the loop increment value instead of just checking each character one by one until we detect that the lines has changed.